backend/akr/src/main/java/fi/oph/akr/service/ClerkTranslatorService.java

```diff
@@ -170,6 +177,28 @@ private List<AuthorisationDTO> createAuthorisationDTOs(final List<AuthorisationP
           .build();
       })
       .toList();
+    if (deduplicateAuthorisations) {
+      return deduplicateAuthorisationDTOs(authorisationDTOS);
+    }
+    return authorisationDTOS;
+  }
+
+  private List<AuthorisationDTO> deduplicateAuthorisationDTOs(final List<AuthorisationDTO> authorisationDTOS) {
+    final Map<LanguagePairDTO, AuthorisationDTO> deduplicated = new LinkedHashMap<>();
+    authorisationDTOS.forEach(a -> {
+      final LanguagePairDTO key = a.languagePair();
+      if (!deduplicated.containsKey(key)) {
+        deduplicated.put(key, a);
+      } else {
+        final AuthorisationDTO existing = deduplicated.get(key);
+        final LocalDate endDate = a.termEndDate();
+        final LocalDate existingEndDate = existing.termEndDate();
+        if (endDate != null && existingEndDate != null && endDate.isAfter(existingEndDate)) {
+          deduplicated.put(key, a);
+        }
```

Comment:
Voiko olla tilannetta jossa `endDate == null || existingEndDate == null` ja pitäisi korvata `deduplicated`in entry? Milkäli on mahdollinen, niin silloin pitäisi kaiketi tarkastella ehtoja hienojakoisemmin.

Reply:
Oikein hyvä kysymys mikä on nyt relevantti kun perusteella ei jaotella. Taitaa olla aika teoreettista, mutta lisäsin ehdon että korvataan myös jos uudella ei ole loppuaikaa ja vanhalla oli (eli VIR korvaa jonkin muun).